integration/client/imagerules/imageroleauthorizations_test.go

```diff
@@ -0,0 +1,358 @@
+
+	// DEPLOY
+
+	// Integration tests don't have proper privileges so we will by pass the permission validation
+	blueprint := &internalv1.AppInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-",
+			Namespace:    c.GetProject(),
+		},
+		Spec: internalv1.AppInstanceSpec{
+			Image: tagName,
+			Permissions: []internalv1.Permissions{
+				{
+					ServiceName: "rootapp",
+					Rules: []internalv1.PolicyRule{{
+						PolicyRule: rbacv1.PolicyRule{
+							APIGroups: []string{"foo.bar.com"},
+							Verbs:     []string{"get"},
+						},
+					}},
+				},
+				{
+					ServiceName: "foo.awsapp",
+					Rules: []internalv1.PolicyRule{{
+						PolicyRule: rbacv1.PolicyRule{
+							APIGroups: []string{"aws.acorn.io"},
+							Verbs:     []string{"get"},
+						},
+					}},
+				},
+			},
+		},
+		Status: internalv1.AppInstanceStatus{},
+	}
+
+	// --------------------------------------------------------------------
+	// Helper Functions
+	// --------------------------------------------------------------------
+	createWaitLoop := func(ctx context.Context, appInstance *internalv1.AppInstance) *apiv1.App {
+		if err := kclient.Create(ctx, appInstance); err != nil {
+			t.Fatal(err)
+		}
+
+		app, err := c.AppGet(ctx, appInstance.Name)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		helper.WaitForObject(t, helper.Watcher(t, c), &apiv1.AppList{}, app, func(obj *apiv1.App) bool {
+			return obj.Status.Staged.PermissionsChecked
+		})
+
+		app, err = c.AppGet(ctx, appInstance.Name)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		return app
+	}
+
+	rmapp := func(ctx context.Context, app *apiv1.App) {
+		_, err := c.AppDelete(ctx, app.Name)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Prep: Create a role that we can use later
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo:bar:admin",
+			Namespace: c.GetNamespace(),
+		},
+		Rules: []rbacv1.PolicyRule{{
+			APIGroups: []string{"foo.bar.com"},
+			Verbs:     []string{"*"},
+			Resources: []string{"*"},
+		}},
+	}
+
+	err = kclient.Create(ctx, role)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// --------------------------------------------------------------------
+	// Run #1 - Expect denied permissions since we have no IRAs
+	// --------------------------------------------------------------------
+	app := createWaitLoop(ctx, blueprint.DeepCopy())
+	require.Equal(t, 1, len(app.Status.Staged.ImagePermissionsDenied), "should have 1 denied permission (rootapp) with no IRA defined")
+	expectedDeniedPermissionsRootapp := []internalv1.Permissions{
+		{
+			ServiceName: tagName,
+			Rules: []internalv1.PolicyRule{
+				{
+					PolicyRule: rbacv1.PolicyRule{
+						APIGroups: []string{"foo.bar.com"},
+						Verbs:     []string{"get"},
+					},
+				},
+			},
+		},
+	}
+	require.Equal(t, expectedDeniedPermissionsRootapp, app.Status.Staged.ImagePermissionsDenied, "should have denied permissions for foo.bar.com on rootapp")
+	rmapp(ctx, app)
```

Comment:
I think this `rmapp` call should be done in a `t.Cleanup` call prior to these `require.Equal` calls. That way the app gets cleaned up if the test fails here.

Reply:
Yeah, I just wanted to remove them as early as possible instead of deferring it to the cleanup.
However, shouldn't this also be handled by the TempProject cleanup?

Reply:
Oh probably, yeah. Good point